website/docs/src/content/cloud/storage/sql-database.mdx

```diff
@@ -0,0 +1,324 @@
+  "TransferFunds",
+  async ({ fromAccount, toAccount, amount }) => {
+    const db = await useDatabase("banking");
+
+    try {
+      // Execute multiple statements as a transaction
+      const result = await db.batch([
+        {
+          sql: "UPDATE accounts SET balance = balance - ? WHERE account_id = ?",
+          params: [amount, fromAccount],
+        },
+        {
+          sql: "UPDATE accounts SET balance = balance + ? WHERE account_id = ?",
+          params: [amount, toAccount],
+        },
+      ]);
+
+      return { success: true, rowsAffected: result.rowsAffected };
+    } catch (error) {
+      return { success: false, error: error.message };
+    }
+  },
+);
+```
+
+### Multi-statement scripts
+
+For complex database changes, you can execute multiple statements at once:
+
+```tsx
+const SetupUserSystem = gensx.Component("SetupUserSystem", async () => {
+  const db = await useDatabase("users");
+
+  // Execute a SQL script with multiple statements
+  await db.executeMultiple(`
+      CREATE TABLE IF NOT EXISTS users (
+        id TEXT PRIMARY KEY,
+        name TEXT NOT NULL,
+        email TEXT UNIQUE
+      );
+
+      CREATE TABLE IF NOT EXISTS user_preferences (
+        user_id TEXT PRIMARY KEY,
+        theme TEXT DEFAULT 'light',
+        notifications BOOLEAN DEFAULT 1,
+        FOREIGN KEY (user_id) REFERENCES users(id)
+      );
+
+      CREATE INDEX IF NOT EXISTS idx_users_email ON users(email);
+    `);
+
+  return "User system set up successfully";
+});
+```
+
+### Database schema migrations
+
+When you need to update your database schema, use migrations:
+
+```tsx
+const MigrateDatabase = gensx.Component(
+  "MigrateDatabase",
+  async ({ version }) => {
+    const db = await useDatabase("app_data");
+
+    if (version === "v2") {
+      // Run migrations with foreign key checks disabled
+      await db.migrate(`
+        ALTER TABLE products ADD COLUMN category TEXT;
+        CREATE TABLE product_categories (
+          id INTEGER PRIMARY KEY,
+          name TEXT NOT NULL,
+          description TEXT
+        );
+      `);
+
+      return "Database migrated to v2";
+    }
+
+    return "No migration needed";
+  },
+);
+```
+
+## Development vs. production
+
+GenSX SQL databases work identically in both local development and cloud environments:
+
+- **Local development**: Databases are stored as SQLite files in the `.gensx/databases` directory by default
+- **Cloud deployment**: Databases are automatically provisioned in the cloud
+
+No code changes are needed when moving from development to production.
+
+## Use cases
+
+### Data-backed agents
+
+Create agents that can query and update structured data:
+
+```tsx
+const DataAnalyst = gensx.Component("DataAnalyst", async ({ query }) => {
+  // Initialize the database with the baseball stats
+  await InitializeDatabase();
+
+  // Use the SQL Copilot to answer the question
+  return <SqlCopilot question={query} />;
+});
+```
```

Comment:
This example feels off because these functions aren't explained. The simplest adjustment might just be to mention that you can do this uses the components defined above